CRM/Event/Cart/Form/Checkout/ParticipantsAndPrices.php

```diff
@@ -184,10 +206,38 @@ public function setDefaultValues() {
         $participant->contact_id = self::find_or_create_contact($this->getContactID());
       }
       $defaults += $form->setDefaultValues();
+      //Set price defaults if any
+      foreach ($this->cart->get_main_events_in_carts() as $event_in_cart) {
+        $price_set_id = CRM_Event_BAO_Event::usesPriceSet($event_in_cart->event_id);
+        if ($price_set_id) {
+          $price_sets = CRM_Price_BAO_PriceSet::getSetDetail($price_set_id, TRUE, TRUE);
+          $price_set  = $price_sets[$price_set_id];
+          foreach ($price_set['fields'] as $field) {
+            $options = CRM_Utils_Array::value('options', $field);
+            if (!is_array($options)) {
+              continue;
+            }
+            $field_name = "event_{$event_id}_price_{$field['id']}";
```

Comment:
If there are any more style issues, you can just attach the Jenkins report.
I also submitted CRM 19001, assigned to myself, to include the Cart form in
cividiscount, so ppl can apply discounts on events to cart items.

On Sat, Jun 25, 2016 at 1:06 PM, Lesley Evensen <zorgalina@gmail.com> wrote:

> What happened - new line 211 sets $event_id in ParticipantsAndPrices.php -
> the snippet from Seamus doesn't have that line? Is that the right file?
>
> I will fix the style issues and re-commit and push.
> Sorry about the } else {  .
> And I will take out the & (pass by ref) of $event_id in apply_discounts
> functions, and all the rest you kindly pointed out
>
> On Sat, Jun 25, 2016 at 1:51 AM, Seamus Lee <notifications@github.com>
> wrote:
>
>> In CRM/Event/Cart/Form/Checkout/ParticipantsAndPrices.php
>> <https://github.com/civicrm/civicrm-core/pull/8427#discussion_r68486416>:
>>
>> > @@ -184,10 +206,38 @@ public function setDefaultValues() {
>> >          $participant->contact_id = self::find_or_create_contact($this->getContactID());
>> >        }
>> >        $defaults += $form->setDefaultValues();
>> > +      //Set price defaults if any
>> > +      foreach ($this->cart->get_main_events_in_carts() as $event_in_cart) {
>> > +        $price_set_id = CRM_Event_BAO_Event::usesPriceSet($event_in_cart->event_id);
>> > +        if ($price_set_id) {
>> > +          $price_sets = CRM_Price_BAO_PriceSet::getSetDetail($price_set_id, TRUE, TRUE);
>> > +          $price_set  = $price_sets[$price_set_id];
>> > +          foreach ($price_set['fields'] as $field) {
>> > +            $options = CRM_Utils_Array::value('options', $field);
>> > +            if (!is_array($options)) {
>> > +              continue;
>> > +            }
>> > +            $field_name = "event_{$event_id}_price_{$field['id']}";
>>
>> Isn't it defined at new line 211?
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/civicrm/civicrm-core/pull/8427/files/b05572f8772dbb5c88434ce9c29a3badc884c286#r68486416>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/AFOn7-fnXtNgEd8kK-6d2z3SChwD50HFks5qPOwcgaJpZM4IkFZ7>
>> .
>>
>
>
>
> --
> زورگلنه ءء لسلی
>



-- 
زورگلنه ءء لسلی